src/runtime/set.cpp

```diff
@@ -114,7 +133,11 @@ Box* frozensetNew(Box* _cls, Box* container) {
 
     // Some optimizations from CPython: frozensets can be shared:
     if (_cls == frozenset_cls) {
-        if (!container) {
+        bool isEmpty = false;
+        if (container && PyObject_HasAttr(container, internStringMortal("__len__"))) {
```

Comment:
@kmod Would you mind to watch the line 598 in new `test_set.py`, there have:
```
frozenset(), frozenset([]), frozenset(()), frozenset(''),
               frozenset(), frozenset([]), frozenset(()), frozenset(''),
               frozenset(xrange(0)), frozenset(frozenset())
```
Need return one empty `frozenset`. This argument could not detect with `None`. I also tried `__nonzero__`, but it is not satisfied `xrange(0)`. From my knowledge, only `__len__` could meet this condition. Maybe there has a better way?